README.md

```diff
@@ -38,191 +38,121 @@ Once [installed](#installation), you should be able to submit custom metrics fro
+The list of instrument services can be found here: https://docs.datadoghq.com/tracing/setup_overview/compatibility_requirements/java/ .
+If you wish to enable a beta integration, please note that you must do so using an environment variable.
 
-```
-INFO 2020-11-11T14:00:00Z LAMBDA_REQUEST_ID [dd.trace_id=12345 dd.span_id=67890] This is a log message
-```
+# Trace/Log Correlation
 
-Then your parser rule would look like:
+Please see [Connecting Java Logs and Traces](https://docs.datadoghq.com/tracing/connect_logs_and_traces/java/?tab=log4j2)
 
-```
-my_custom_rule \[%{word:level}\]?\s+%{_timestamp}\s+%{notSpace:lambda.request_id}%{_trace_context}?.*
-```
+In brief, if you set the environment variable `DD_LOGS_INJECTION=true`, your trace ID and span ID are automatically injected into the MDC.
+If you are using JSON-formatted logs and logging using Logback, there is nothing left to do.
 
-#### Log4j / SLF4J
+## Raw-formatted logs (Log4J, etc.)
 
-We have added the Trace ID into the slf4j MDC under the key `dd.trace_context`. That can be accessed
-using the `%X{dd.trace_context}` operator. Here is an example `log4j.properties`: 
+For raw formatted logs, you must update your log format and your log parser. Instructions for both are below.
 
-```
-log = .
-log4j.rootLogger = DEBUG, LAMBDA
+### Log Format
 
-log4j.appender.LAMBDA=com.amazonaws.services.lambda.runtime.log4j.LambdaAppender
-log4j.appender.LAMBDA.layout=org.apache.log4j.PatternLayout
-log4j.appender.LAMBDA.layout.conversionPattern=%d{yyyy-MM-dd HH:mm:ss} %X{dd.trace_context} %-5p %c:%L - %m%n
+If you are using raw formatted logs, update your formatter to include `dd.trace_context`. E.g.
+
+```xml
+<Pattern>"%d{yyyy-MM-dd HH:mm:ss} <%X{AWSRequestId}> %-5p %c:%L %X{dd.trace_context} %m%n"</Pattern>
+<!--Please note:                      Request ID                      Trace Context  -->
 ```
 
-would result in log lines looking like `2020-11-13 19:21:53 [dd.trace_id=1168910694192328743 dd.span_id=3204959397041471598] INFO  com.serverless.Handler:20 - Test Log Message`
+Please note that `RequestId` has also been added. 
+RequestId is not strictly necessary for Trace/Log correlation, but it is useful for correlating logs and invocations.
 
-Just like the **Plain Text Logs** in the previous section, you must create a new [Parser](https://docs.datadoghq.com/logs/processing/parsing/?tab=matcher) in order to parse the trace context correctly.
 
+### Grok Parser
 
-#### Other logging solutions
+The following grok parser parses Java logs formatted using the pattern in the previous section.
 
-If you are using a different logging solution, the trace ID can be accessed using the method
-`DDLambda.getTraceContextString()`. That returns your trace ID as a string that can be added
-to any log message.
+```
+java_tracer %{date("yyyy-MM-dd HH:mm:ss"):timestamp}\s\<%{uuid:lambda.request_id}\>\s%{word:level}\s+%{data:call_site}%{_trace_context}%{data:message}
+```
```

Comment:
Is this parser the same one we provide out of the box with the default Lambda log pipeline? If that is the case, shouldn't this just work without the customer having to know about the parser? If that is **not** the case, is there a reason we can't format the logs in the format already understood by the default parser?

Reply:
We don't have the power (as far as I know) to change how the customer formats their logs. We can just suggest formats. I think the smart thing to do is to bias towards action, get this out there, and see how customers use it.